### PR TITLE
[Doc] Add beginner mode to hide advanced doc chapters

### PR DESCRIPTION
**Pull request by fzaninotto**
## Problem

React-admin is a large framework, and its doc may be overwhelming for beginners

## Solution

- [x] Display a button allowing to hide advanced doc chapters
- [x] Mention this beginner mode in the tutorial

https://github.com/marmelab/react-admin/assets/99944/15b2350f-db6f-47b1-aa48-5e91759b7324

### CODE REVIEW

File: docs/navigation.html [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
+</div>
+
 <ul><div>Getting Started</div>
-  <li {% if page.path == 'Tutorial.md' %} class="active" {% endif %}><a class="nav-link" href="./Tutorial.html">Tutorial</a></li>
-  <li {% if page.path == 'Features.md' %} class="active" {% endif %}><a class="nav-link" href="./Features.html">Features</a></li>
+  <li {% if page.path == 'Tutorial.md' %} class="active beginner" {% else %} class="beginner" {% else %} class="beginner" {% endif %}><a class="nav-link" href="./Tutorial.html">Tutorial</a></li>
+  <li {% if page.path == 'Features.md' %} class="active beginner" {% else %} class="beginner" {% endif %}><a class="nav-link" href="./Features.html">Features</a></li>
   <li {% if page.path == 'Architecture.md' %} class="active" {% endif %}><a class="nav-link" href="./Architecture.html">Key concepts</a></li>
   <li><a class="nav-link external" href="https://github.com/marmelab/react-admin/releases" target="_blank">What's new?</a></li>
   <li {% if page.path == 'CreateReactAdmin.md' %} class="active" {% endif %}><a class="nav-link" href="./CreateReactAdmin.html">Create React Admin</a></li>
```
Comment (slax57) [inline docs/navigation.html:13]: To me this should be marked 'beginner' as well

File: docs/navigation.html [hunk 2]
```diff
@@ -124,84 +131,84 @@
...
+  <li {% if page.path == 'DateInput.md' %} class="active beginner" {% else %} class="beginner" {% endif %}><a class="nav-link" href="./DateInput.html"><code>&lt;DateInput></code></a></li>
   <li {% if page.path == 'DateTimeInput.md' %} class="active" {% endif %}><a class="nav-link" href="./DateTimeInput.html"><code>&lt;DateTimeInput&gt;</code></a></li>
   <li {% if page.path == 'DualListInput.md' %} class="active" {% endif %}><a class="nav-link" href="./DualListInput.html"><code>&lt;DualListInput&gt;</code><img class="premium" src="./img/premium.svg" /></a></li>
   <li {% if page.path == 'FileInput.md' %} class="active" {% endif %}><a class="nav-link" href="./FileInput.html"><code>&lt;FileInput&gt;</code></a></li>
-  <li {% if page.path == 'ImageInput.md' %} class="active" {% endif %}><a class="nav-link" href="./ImageInput.html"><code>&lt;ImageInput&gt;</code></a></li>
+  <li {% if page.path == 'ImageInput.md' %} class="active beginner" {% else %} class="beginner" {% endif %}><a class="nav-link" href="./ImageInput.html"><code>&lt;ImageInput&gt;</code></a></li>
```
Comment (slax57) [inline docs/navigation.html:193]: This is advanced IMHO

File: docs/navigation.html [hunk 3]
```diff
@@ -124,84 +131,84 @@
...
-  <li {% if page.path == 'ReferenceManyInput.md' %} class="active" {% endif %}><a class="nav-link" href="./ReferenceManyInput.html"><code>&lt;ReferenceManyInput&gt;</code><img class="premium" src="./img/premium.svg" /></a></li>
+  <li {% if page.path == 'RadioButtonGroupInput.md' %} class="active beginner" {% else %} class="beginner" {% endif %}><a class="nav-link" href="./RadioButtonGroupInput.html"><code>&lt;RadioButtonGroupInput&gt;</code></a></li>
+  <li {% if page.path == 'ReferenceInput.md' %} class="active beginner" {% else %} class="beginner" {% endif %}><a class="nav-link" href="./ReferenceInput.html"><code>&lt;ReferenceInput&gt;</code></a></li>
+  <li {% if page.path == 'ReferenceArrayInput.md' %} class="active beginner" {% else %} class="beginner" {% endif %}><a class="nav-link" href="./ReferenceArrayInput.html"><code>&lt;ReferenceArrayInput&gt;</code></a></li>
+  <li {% if page.path == 'ReferenceManyInput.md' %} class="active beginner" {% else %} class="beginner" {% endif %}><a class="nav-link" href="./ReferenceManyInput.html"><code>&lt;ReferenceManyInput&gt;</code><img class="premium" src="./img/premium.svg" /></a></li>
```
Comment (slax57) [inline docs/navigation.html:202]: This is advanced IMHO

**Review by slax57 — COMMENTED** [review]
One last comment: I believe the UI for the checkbox is not particularly beautiful, and does not fit very well with the current design.

image

A bit of CSS might be welcome :slightly_smiling_face:

**Review by slax57 — COMMENTED** [review]
Works nice, well done!

However, I'm still not really found of the color of the switch

image

Can't you make it more in the tones of the rest of the documentation's theme?

**Comment by fzaninotto** [comment]
> Can't you make it more in the tones of the rest of the documentation's theme?

Yep, done.

**Review by djhi — APPROVED** [review]

File: docs/css/style-v18.css [hunk 4]
```diff
@@ -164,6 +164,20 @@ ul.sidenav code {
     box-shadow: none;
 }
 
+.switch label input[type=checkbox]:checked+.lever {
```
Comment (slax57) [inline docs/css/style-v18.css:167]: The new style looks really nice, thanks.

But... shouldn't you create a `style-v19.css` file instead? :roll_eyes:

**Review by slax57 — CHANGES_REQUESTED** [review]

File: docs/css/style-v18.css (hunk 4 above)
Comment (fzaninotto) [reply docs/css/style-v18.css:167]: You're right! It's now fixed.

**Review by slax57 — APPROVED** [review]

**Pull request merged**